### ISSUE TITLE
unused state variable are not detected by no-unused-vars

### ISSUE THREAD

**Issue by rokso** [protofire/solhint#297]
This may or may not be intended behavior. In case it is intended behavior then how to detect unused state variables?

**Comment by dbale-altoros** [comment]
@rokso can you post an example ?

**Comment by rokso** [comment]
@dbale-altoros Does this count? No warning when I run solhint check.

```
pragma solidity 0.8.9;

contract Example {
    address public unusedStateVariable;
}
```

**Comment by dbale-altoros** [comment]
and when you run solhint with the unused var rule active , it doesn't complain ?
is it ignoring that file ? sorry the dumb question I just want to make sure

**Comment by rokso** [comment]
I think rule is active by default. I have manually added it `"no-unused-vars": "error",` and no error. It is not ignoring file.

**Comment by rokso** [comment]
Here is my solhint.json file
```
{
  "extends": ["solhint:recommended"],
  "rules": {
    "prettier/prettier": "error",
    "no-unused-vars": "error",
    "quotes": ["error", "double"],
    "compiler-version": "off",
    "func-visibility": [
      "warn",
      {
        "ignoreConstructors": true
      }
    ]
  },
  "plugins": ["prettier"]
}

```

**Comment by dbale-altoros** [comment]
ok, thanks for the feedback... we will be supporting an update of this project soon

**Comment by rokso** [comment]
@dbale-altoros thanks a ton.

**Comment by dbale-altoros** [comment]
@rokso the thing with your proposal is this:
solhint is not yet suitable to make crosschecks between contracts
very often state variables from a contract are inherited into another one... so it will produce a false positive when a state variable is not being used in the current contract but is used in the one that inherits such variable

I think this is why the rule does not check that
@fvictorio  ?
